### DISCUSSION TITLE
Error message "modem cannot be put in programming mode"

### DISCUSSION THREAD

**Discussion by telit-fabiopo** [fwupd/fwupd#5855] [Q&A]
I'm adding support for a new modem trough the ModemManager plugin.
I'm stuck with this error "modem cannot be put in programming mode", I don't understand the reason for this error and the way to fix it.

Any help will be appreciated.

**Comment by hughsie** [comment]
This is from here:

	self->update_methods = mm_firmware_update_settings_get_method(update_settings);
	if (self->update_methods == MM_MODEM_FIRMWARE_UPDATE_METHOD_NONE) {
		g_set_error_literal(error,
				    FWUPD_ERROR,
				    FWUPD_ERROR_NOT_SUPPORTED,
				    "modem cannot be put in programming mode");
		return FALSE;
	}

i.e. ModemManager doesn't know how to put the device into the correct mode.

**Reply by telit-fabiopo** [reply]
Thanks for the help.

After i modified the ModemManager the error is gone.
